src/plugins/discover/public/application/components/context_app/context_app.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { Fragment, memo, useEffect, useRef, useMemo, useCallback } from 'react';
+import './context_app.scss';
+import classNames from 'classnames';
+import { FormattedMessage, I18nProvider } from '@kbn/i18n/react';
+import { EuiText, EuiPageContent, EuiPage, EuiSpacer } from '@elastic/eui';
+import { cloneDeep } from 'lodash';
+import { esFilters, SortDirection } from '../../../../../data/public';
+import { DOC_TABLE_LEGACY, SEARCH_FIELDS_FROM_SOURCE } from '../../../../common';
+import { ContextErrorMessage } from '../context_error_message';
+import { IndexPattern, IndexPatternField } from '../../../../../data/common';
+import { LoadingStatus } from '../../angular/context_query_state';
+import { getServices } from '../../../kibana_services';
+import { AppState, isEqualFilters } from '../../angular/context_state';
+import { useDataGridColumns } from '../../helpers/use_data_grid_columns';
+import { useContextAppState } from './use_context_app_state';
+import { useContextAppFetch } from './use_context_app_fetch';
+import { popularizeField } from '../../helpers/popularize_field';
+import { ContextAppContent } from './context_app_content';
+
+const ContextAppContentMemoized = memo(ContextAppContent);
+
+export interface ContextAppProps {
+  indexPattern: IndexPattern;
+  indexPatternId: string;
+  anchorId: string;
+}
+
+export const ContextApp = ({ indexPattern, indexPatternId, anchorId }: ContextAppProps) => {
+  const services = getServices();
+  const { uiSettings: config, capabilities, indexPatterns, navigation, filterManager } = services;
+
+  const isLegacy = useMemo(() => config.get(DOC_TABLE_LEGACY), [config]);
+  const useNewFieldsApi = useMemo(() => !config.get(SEARCH_FIELDS_FROM_SOURCE), [config]);
```

Comment:
Are there really performance optimizations in using `memo` here?

Reply:
In case of omitting memo usage here, on each anchor id change `useNewFieldsApi` and `is Legacy` settings will be retrived once more